### ISSUE TITLE
Installing a tar distro in wsl 1 with --from-file won't run the OOBE

### ISSUE THREAD

**Issue by crramirez** [microsoft/WSL#12398]
### Windows Version

Microsoft Windows [Version 10.0.26100.2605]

### WSL Version

2.4.8.0

### Are you using WSL 1 or WSL 2?

- [ ] WSL 2
- [x] WSL 1

### Repro Steps

```powershell
wsl --install --from-file distro.tar --version 1
wsl -d distroName

```

### Expected Behavior

Run the OOBE

### Actual Behavior

Logs directly with root:

![Image](https://github.com/user-attachments/assets/e4fa502c-f03e-4a0f-b50e-9c4f49dea71d)

**Comment by elsaco** [comment]
The guide at https://learn.microsoft.com/en-us/windows/wsl/build-custom-distro states that `This guide only applies to WSL release 2.4.4 and higher`, so is WSL1 aware of the  `[oobe]` section in `etc/wsl-distribution.conf`?

**Comment by CarlosNihelton** [comment]
I also observed this behaviour with WSL 2.4.9, so I suspect it's a bug.

**Comment by arixmkii** [comment]
Can confirm with WSL 2.4.9. I need to open shell with `wsl -d distroname` for oobe to run.

**Comment by crramirez** [comment]
@arixmkii this is the expected behavior. But in WSL1 it won't run the oobe even after you open the shell with `wsl -d distroname`

But I agree with you that it should be executed automatically to mimic the former behavior.

**Comment by arixmkii** [comment]
Yeah, I gave it additional thinking. It is logical a bit, because oobe might require interactive manipulations, so, it would be a bad default. May be a command flag to force run oobe: --install --from-file filename --with-oobe

Opening terminal works for me - just had to get used to it.

**Comment by elsaco** [comment]
In `wsl-2.4.8` when adding a new modern distro the script `oobe.sh` is run on the first launch:

![Image](https://github.com/user-attachments/assets/0b842328-3982-425b-86a5-d88fe737851b)

After installing the new distro and before launching it look for the `RunOOBE` registry key in `HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss` for that specific distro. It should be set to `0x1`

**Comment by CarlosNihelton** [comment]
With WSL 2.4.10 I got the OOBE running by default as expected.

**Comment by crramirez** [comment]
Fixed in: https://github.com/microsoft/WSL/releases/tag/2.4.10
